crates/ott-balancer/src/balancer.rs

```diff
@@ -179,6 +179,65 @@ pub struct BalancerContext {
     pub monoliths_by_region: HashMap<String, Vec<MonolithId>>,
 }
 
+pub struct MonolithRegistry {
+    pub monoliths_by_region: HashMap<String, Vec<MonolithId>>,
+    pub monoliths: HashMap<MonolithId, BalancerMonolith>,
+}
```

Comment:
Ideally, this would be a zero size type with no members. These maps should remain in `BalancerContext` and not cloned.

Reply:
I made this change. Since `select_monolith` and `random_monolith` rely on `monoliths` and `monoliths_by_region` defined in `BalancerContext` for the load balancing logic, how do we make these available to the new struct without cloning them? 

That's why I asked that second question, because I knew we shouldn't clone those two members, but the function logic depends on them. I'm taking a look at `behavior.rs` right now.

Reply:
>  without cloning them

By borrowing them, as references.